zookeeper-api/src/main/java/org/apache/helix/zookeeper/impl/client/ZkClient.java

```diff
@@ -187,6 +197,16 @@ public static class Builder {
     String _monitorKey;
     String _monitorInstanceName = null;
     boolean _monitorRootPathOnly = true;
+    boolean _connectOnInit = true;
+
+    /**
+     * If set true, the client will connect to ZK during initialization.
+     * Otherwise, user has to call connect() method explicitly before talking to ZK.
+     */
+    public Builder setConnectOnInit(boolean connectOnInit) {
```

Comment:
since this is during init, will this method be of any use?

Reply:
Sorry, I don't get it. This is the builder class, not zkclient itself.